patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -1350,36 +1350,80 @@
  		}
  
  		UpdateHungerBuffs();
-@@ -9241,7 +_,7 @@
+@@ -9226,23 +_,26 @@
+ 		accStopwatch = false;
+ 		accOreFinder = false;
+ 		accDreamCatcher = false;
++
++		PlayerLoader.ResetInfoAccessories(this);
++
+ 		for (int i = 0; i < 58; i++) {
+ 			int type = inventory[i].type;
+-			RefreshInfoAccsFromItemType(type);
++			RefreshInfoAccsFromItemType(inventory[i]);
+ 			if (type == 4131)
+ 				flag = true;
+ 		}
+ 
+ 		for (int j = 0; j < 10; j++) {
+ 			int type2 = armor[j].type;
+-			RefreshInfoAccsFromItemType(type2);
++			RefreshInfoAccsFromItemType(armor[j]);
+ 		}
+ 
  		if (flag) {
  			for (int k = 0; k < 40; k++) {
  				int type3 = bank4.item[k].type;
 -				if (type3 < 0 || type3 > ItemID.Count || ItemID.Sets.WorksInVoidBag[type3])
+-					RefreshInfoAccsFromItemType(type3);
 +				if (type3 < 0 || ItemID.Sets.WorksInVoidBag[type3])
- 					RefreshInfoAccsFromItemType(type3);
++					RefreshInfoAccsFromItemType(bank4.item[k]);
  			}
  		}
-@@ -9431,6 +_,9 @@
+ 
+@@ -9301,8 +_,17 @@
+ 
+ 				if (Main.player[i].hasLuck_LuckyCoin)
+ 					hasLuck_LuckyCoin = true;
++
++				PlayerLoader.RefreshInfoAccessoriesFromTeamPlayers(this, Main.player[i]);
+ 			}
+ 		}
++	}
++
++	// Added by tML.
++	public void RefreshInfoAccsFromItemType(Item item)
++	{
++		RefreshInfoAccsFromItemType(item.type);
++		ItemLoader.UpdateInfoAccessory(item, this);
+ 	}
+ 
+ 	public void RefreshInfoAccsFromItemType(int accType)
```

Comment:
make `private` please, since it doesn't call the hook. Alternatively, no need to actually split the method. Just declare `int accType`